### ISSUE TITLE
Falcon Implementation

### ISSUE THREAD

**Issue by gmuraru** [OpenMined/PySyft#3715]
## Description
Implement [Falcon](https://arxiv.org/abs/2004.02229).

## Why?
For the moment we have implemented the SecureNN - [github](https://github.com/snwagh/securenn-public), [paper](https://eprint.iacr.org/2018/442.pdf) which leverages the power of SMPC to be able to run training/inference algorithms, but Falcon promises 8x faster computation speed.

## Breakdown
Those are mainly breakdown of subsections presented in the paper:
- [ Basic Operations](https://github.com/OpenMined/PySyft/issues/3852)
- [Private Compare](https://github.com/OpenMined/PySyft/issues/3967)
- [Wrap Function](https://github.com/OpenMined/PySyft/issues/3969)
- [ReLU and Derivative of ReLU](https://github.com/OpenMined/PySyft/issues/3970)
- [Maxpool and Derivative of Maxpool](https://github.com/OpenMined/PySyft/issues/3971)
- [Division and Batch Normalization](https://github.com/OpenMined/PySyft/issues/3972)
## Milestone
https://github.com/OpenMined/PySyft/milestone/14 
## Who else?
This work should involve/affect the @OpenMined/cryptography team.

**Comment by ibarrond** [comment]
Is this implementation still a thing? I see all issues in the milestone marked as "stale".
